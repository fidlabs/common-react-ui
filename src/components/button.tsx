import type { HTMLAttributes } from "react";

export type ButtonProps = HTMLAttributes<HTMLButtonElement>;

export function Button({ children, ...rest }: ButtonProps) {
  return <button {...rest}>{children}</button>;
}
