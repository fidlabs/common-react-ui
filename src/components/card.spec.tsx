import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import { Button } from "./button";
import {
  Card,
  CardContent,
  CardDescription,
  CardFooter,
  CardHeader,
  CardTitle,
} from "./card";

describe("Card component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <Card>
        <CardHeader>
          <CardTitle>Card Title</CardTitle>
          <CardDescription>Card Description</CardDescription>
        </CardHeader>

        <CardContent>
          <p>Text in the card content.</p>
        </CardContent>

        <CardFooter>
          <Button>Footer Button</Button>
        </CardFooter>
      </Card>
    );
    expect(container).toMatchSnapshot();
  });
});
