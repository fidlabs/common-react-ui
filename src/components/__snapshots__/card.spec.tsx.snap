// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card component > renders correctly 1`] = `
<div>
  <div
    class="rounded-lg border-none text-card-foreground shadow-f-card bg-white"
  >
    <div
      class="p-6 w-full"
    >
      <h3
        class="text-lg font-semibold leading-none tracking-tight text-primary"
      >
        Card Title
      </h3>
      <p
        class="text-sm text-muted-foreground"
      >
        Card Description
      </p>
    </div>
    <div
      class="p-6 pt-0"
    >
      <p>
        Text in the card content.
      </p>
    </div>
    <div
      class="flex items-center p-6 pt-0"
    >
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 gap-2 bg-dodger-blue text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
      >
        Footer Button
      </button>
    </div>
  </div>
</div>
`;
